#!/usr/bin/env python
PACKAGE = "visual_servoing_davis"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("eHarris_threshold", double_t, 0, " The inverse ratio of resolution",    5, 0,   10)
gen.add("eHarris_window_size", int_t, 0, " The inverse ratio of resolution",    4, 0,   25)
gen.add("eHarris_queue_size", int_t, 0, " The inverse ratio of resolution",    30, 0,   50)
gen.add("Harris_threshold", int_t, 0, " The inverse ratio of resolution",    70, 0,   255)
gen.add("Harris_block_size", int_t, 0, " The inverse ratio of resolution",    5, 0,   10)
gen.add("Harris_aperture_size", int_t, 0, " The inverse ratio of resolution",    7, 0,   10)
gen.add("target_velocity", double_t, 0, " The inverse ratio of resolution",    0.1, 0,   0.2)
gen.add("heatmap_time_decay_factor", double_t, 0, " The inverse ratio of resolution",    0.1, 0,   1)
gen.add("heatmap_corner_weight", double_t, 0, " The inverse ratio of resolution",    0.15, 0,   1)
gen.add("heatmap_threshold", double_t, 0, " The inverse ratio of resolution",    0.6, 0,   1)

exit(gen.generate(PACKAGE, "visual_servoing_davis", "VSCfg"))
